pinecone/index_connection.go

```diff
@@ -996,8 +1003,307 @@ func (idx *IndexConnection) DescribeIndexStatsFiltered(ctx context.Context, meta
+//   - uri: The URI of the data to import. The URI must start with the scheme of a supported storage provider.
+//   - integrationId: If your bucket requires authentication to access, you need to pass the id of your storage integration using this property.
+//     Pass nil if not required.
+//   - errorMode: If set to "continue", the import operation will continue even if some records fail to import.
+//     Pass "abort" to stop the import operation if any records fail. Will default to "continue" if nil is passed.
+//
+// Example:
+//
+//		 ctx := context.Background()
+//
+//		 clientParams := pinecone.NewClientParams{
+//		     ApiKey:    "YOUR_API_KEY",
+//			 SourceTag: "your_source_identifier", // optional
+//	     }
+//
+//	     pc, err := pinecone.NewClient(clientParams)
+//	     if err != nil {
+//		     log.Fatalf("Failed to create Client: %v", err)
+//	     }
+//
+//	     idx, err := pc.DescribeIndex(ctx, "your-index-name")
+//	     if err != nil {
+//		     log.Fatalf("Failed to describe index \"%s\". Error:%s", idx.Name, err)
+//	     }
+//
+//	     idxConnection, err := pc.Index(pinecone.NewIndexConnParams{Host: idx.Host})
+//	     if err != nil {
+//		     log.Fatalf("Failed to create IndexConnection for Host: %v. Error: %v", idx.Host, err)
+//	     }
+//
+//	     uri := "s3://your-bucket/your-file.csv"
+//	     errorMode := "abort"
+//	     importRes, err := idxConnection.StartImport(ctx, uri, nil, &errorMode)
+//	     if err != nil {
+//	         log.Fatalf("Failed to start import: %v", err)
+//	     }
+//	     fmt.Printf("import starteed with ID: %s", importRes.Id)
+func (idx *IndexConnection) StartImport(ctx context.Context, uri string, integrationId *string, errorMode *ImportErrorMode) (*StartImportResponse, error) {
+	if uri == "" {
+		return nil, fmt.Errorf("must specify a uri to start an import")
+	}
+
+	req := db_data_rest.StartImportRequest{
+		Uri:           &uri,
+		IntegrationId: integrationId,
+	}
+
+	if errorMode != nil {
+		req.ErrorMode = &db_data_rest.ImportErrorMode{
+			OnError: pointerOrNil(db_data_rest.ImportErrorModeOnError(*errorMode)),
+		}
+	}
+
+	res, err := (*idx.restClient).StartBulkImport(idx.akCtx(ctx), req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, handleErrorResponseBody(res, "failed to start import: ")
+	}
+
+	return decodeStartImportResponse(res.Body)
+}
+
+// DescribeImport retrieves information about a specific import operation.
+//
+// Returns an Import object representing the current state of the import or an error if the request fails.
+//
+// Parameters:
+//   - ctx: A context.Context object controls the request's lifetime,
+//     allowing for the request to be canceled or to timeout according to the context's deadline.
+//   - id: The id of the import operation. This is returned when you call StartImport, or can be retrieved
+//     through the ListImports method.
```

Comment:
Can you link to other funcs in the docstrings of Go funcs? Might be nice to link to `ListImports` here

Reply:
Good call, updated to link to the explicit methods. I think we could do more of this elsewhere, I think we've just been using proper names in the doc comments until now. Added a chore ticket to go through these: https://app.asana.com/0/1203260648987893/1208561704179017/f

Reply:
Sweet sounds great, thanks!